src/cobra/core/reaction.py

```diff
@@ -1041,20 +1167,20 @@ def _dissociate_gene(self, cobra_gene):
         self._genes.discard(cobra_gene)
         cobra_gene._reaction.discard(self)
 
-    def knock_out(self):
+    def knock_out(self) -> None:
         """Knockout reaction by setting its bounds to zero."""
         self.bounds = (0, 0)
 
     def build_reaction_from_string(
         self,
-        reaction_str,
-        verbose=True,
-        fwd_arrow=None,
-        rev_arrow=None,
-        reversible_arrow=None,
-        term_split="+",
-    ):
-        """Builds reaction from reaction equation reaction_str using parser
+        reaction_str: str,
+        verbose: Optional[bool] = True,
+        fwd_arrow: Optional[AnyStr] = None,
```

Comment:
I believe `re.compile()` generates a regular expression object and `AnyStr` is not accurate for that.

Reply:
Okay, yes you are correct. My bad, I missed the usage part. Let's keep it `AnyStr`.